### PR TITLE
[Validator] Tidy-up of consolidated staking amount calculator

### PR DESCRIPTION
**Pull request by hyunsooda**
## Proposed changes

The current implementation readability of the consolidated staking calculator is quite poor. This PR aims to improve readability by tidying up the code to a minimal and more easily understandable state. This PR was inspired by the [@ian0371's comment](https://github.com/Krustuniverse-Klaytn-Group/trustless-bridge-poc/pull/19#discussion_r1209672405)

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: consensus/istanbul/validator/weighted.go [hunk 1]
```diff
@@ -752,43 +752,28 @@ func filterValidators(isSingleMode bool, govNodeAddr common.Address, weightedVal
...
-		if err == nil {
-			rewardAddr := stakingInfo.CouncilRewardAddrs[sIdx]
-			weightedVal.SetRewardAddress(rewardAddr)
-			stakingAmounts[vIdx] = float64(stakingInfo.CouncilStakingAmounts[sIdx])
-			addedStaking[sIdx] = true
-		} else {
-			weightedVal.SetRewardAddress(common.Address{})
-		}
-	}
+		weightedVals[vIdx] = weightedVal
```
Comment (jeongkyun-oh) [inline consensus/istanbul/validator/weighted.go:764]: @hyunsooda It seems that `SetRewardAddress` calls disappear.  Don't we need to call them?

Comment (hyunsooda) [reply consensus/istanbul/validator/weighted.go:764]: Great finding! This lacks can forward block rewards to `REWARDBASE` account, not the registered reward address to the CnStaking contract. Updated

**Comment by Sotatek-TinnNguyen** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by aidan-kwon** [comment]
@hyunsooda I am gonna merge this PR. If you have any concern about this, please let me know

**Pull request merged**